src/components/BillFilter/BillFilter.jsx

```diff
@@ -1,227 +1,167 @@
  */
 const BillFilter = ({
-  dateFilters,
-  filter,
+  activeTab,
+  applyFiltering,
+  currentSorting,
   billStatusFilter,
+  closeHandler,
+  dateCheckboxHandler,
+  dateFilters,
+  filterToggles,
+  filterToggleChange,
   handleDateSelection,
-  updateBills,
-  setState,
+  handleStatusChange,
 }) => {
+  const wrapperRef = useRef(null);
+  useOutsideAlerter(wrapperRef, closeHandler);
+
   const filterBadges = [
-    { id: "descending", title: "Newest", onClick: () => updateBills("Newest") },
-    { id: "ascending", title: "Oldest", onClick: () => updateBills("Oldest") },
+    {
+      id: "descending",
+      title: "Newest",
+      onClick: () => applyFiltering("Newest"),
+    },
+    {
+      id: "ascending",
+      title: "Oldest",
+      onClick: () => applyFiltering("Oldest"),
+    },
     {
       id: "by__status",
       title: "Status",
       onClick: () =>
-        setState({
-          filter: {
-            ...filter,
-            statusOpened: true,
-            categoryOpened: false,
-            dateOpened: false,
-          },
-          dateFilters: {
-            ...dateFilters,
-            startDate: { ...dateFilters.startDate, selected: false },
-            endDate: { ...dateFilters.endDate, selected: false },
-          },
+        filterToggleChange({
+          statusOpened: true,
+          categoryOpened: false,
+          dateOpened: false,
         }),
     },
     {
       id: "alphabetical__increasing",
       title: "A to Z",
-      onClick: () => updateBills("A to Z"),
+      onClick: () => applyFiltering("A to Z"),
     },
     {
       id: "alphabetical__decreasing",
       title: "Z to A",
-      onClick: () => updateBills("Z to A"),
+      onClick: () => applyFiltering("Z to A"),
     },
     {
       id: "by__category",
       title: "Category",
       onClick: () =>
-        setState({
-          filter: {
-            ...filter,
-            categoryOpened: true,
-            statusOpened: false,
-            dateOpened: false,
-          },
-          dateFilters: {
-            ...dateFilters,
-            startDate: { ...dateFilters.startDate, selected: false },
-            endDate: { ...dateFilters.endDate, selected: false },
-          },
+        filterToggleChange({
+          categoryOpened: true,
+          statusOpened: false,
+          dateOpened: false,
         }),
     },
     {
       id: "by__date",
       title: "Date",
       onClick: () =>
-        setState({
-          filter: {
-            ...filter,
-            dateOpened: true,
-            categoryOpened: false,
-            statusOpened: false,
-          },
+        filterToggleChange({
+          dateOpened: true,
+          categoryOpened: false,
+          statusOpened: false,
         }),
     },
   ];
 
   const extraFilterSelected =
-    filter.dateOpened === true ||
-    filter.statusOpened === true ||
-    filter.categoryOpened === true;
-
+    filterToggles.dateOpened === true ||
```

Comment:
is the reason we're doing === here because dateOpened could be other values and we need to check that they're boolean? 

Otherwise why can't we just do dateOpened || statusOpened || categoryOpened

Reply:
We can most definitely do that.